### PR TITLE
Diags: <diags>.write_fields Switch

### PR DESCRIPTION
**Pull request by ax3l**
As with `<diags>.write_species`, this switch can be used to disable field writes in diagnostics. This is useful to dump particle species and field data with different intervals.

Note:
Plotfiles need grid meta data to read in particles.

### CODE REVIEW

**Review by RevathiJambunathan — APPROVED** [review]
Thanks for this PR!

**Review by MaxThevenet — APPROVED** [review]
Thanks for this PR!

File: Source/Diagnostics/Diagnostics.cpp [hunk 1]
```diff
@@ -43,6 +43,18 @@ Diagnostics::BaseReadParameters ()
...
     }
 
+    // default for writing field output is 1
+    int write_fields = 1;
+    pp.query("write_fields", write_fields);
+    if (write_fields == 0) {
+        if (m_format == "checkpoint") {
+            amrex::Abort("For checkpoint format, write_fields flag must be 1.");
+        }
+        // if user-defined value for write_fields == 0, then clear species vector
```
Comment (lucafedeli88) [inline Source/Diagnostics/Diagnostics.cpp]:
```suggestion
        // if user-defined value for write_fields == 0, then clear field vector
```
I guess that `species vector` should be changed with something like this

**Review by lucafedeli88 — APPROVED** [review]
Great! I left only one minor comment!

**Comment by ax3l** [comment]
Thank y'all! Actually, it looks like there is a bug to spot here. Could not see the same crash locally yet, hides a bit.

**Comment by ax3l** [comment]
We should finalize this and remove the `do_back_transformed_fields/do_back_transformed_particles` options to avoid confusion. X-ref: #4089

File: Source/Diagnostics/Diagnostics.cpp [hunk 2]
```diff
@@ -72,6 +72,31 @@ Diagnostics::BaseReadParameters ()
...
+    pp.query("write_fields", write_fields);
+    if (write_fields == 0) {
+        if (m_format == "checkpoint") {
+            ablastr::warn_manager::WMRecordWarning(
+                "Diagnostics",
+                "For checkpoint format, write_fields flag must be 1. Overwriting."
+            );
+            pp.add("write_fields", 1);
+        }
+        else if (m_format == "plotfile") {
```
Comment (ax3l) [inline Source/Diagnostics/Diagnostics.cpp:86]: Technically, we can probably make this: if plotfile AND lab diagnostics.

For BTD, this works. We can unify the flags in a follow-up PR.

**Comment by dpgrote** [comment]
Is the `write_fields` option duplicative? There is already the option `fields_to_plot` which can be set to `none` to turn off the writing of the fields. Does BTD use the `fields_to_plot` option?

**Comment by RemiLehe** [comment]
As mentioned by @dpgrote, this capability already exists when setting `fields_to_plot = none`

**Pull request closed without merging**